dragonfly/implementations/at_operator.py

```diff
@@ -0,0 +1,470 @@
+        elif shift_end_time > self.current_shift_end_time:
+            self.current_shift_end_time = shift_end_time
+            logger.info('Extended the shift time for current operator!')
+            message = "It seems that the shift for current operator is extended! \n"
+            message += "I'm not sure whether this branch will be used though."
+        if not regular_check:
+            self.send_message(self.monitor_channel_id, message)
+
+    def construct_user_dictionaries(self):
+        '''
+        Return 3 dictionaries storing information of Slack users: full name to id, id to username, and username to id.
+        '''
+        request = self.slack_client.api_call("users.list")
+        if request['ok']:
+            full_name_to_id_dictionary = {}
+            id_to_username_dictionary = {}
+            username_to_id_dictionary = {}
+            for m in request['members']:
+                if not m['is_bot'] and not m['deleted']:
+                    full_name_to_id_dictionary[m['real_name']] = m['id']
+                    id_to_username_dictionary[m['id']] = m['name']
+                    username_to_id_dictionary[m['name']] = m['id']
+            logger.info('Constructed dictionaries for full names, Slack usernames and ids.')
+            return full_name_to_id_dictionary, id_to_username_dictionary, username_to_id_dictionary
+        else:
+            return None, None, None
+
+    def construct_channel_dictionaries(self):
+        '''
+        Return 2 dictionaries storing information of Slack channels: channel name to id, and channel id to name.
+        '''
+        logger.info("Trying to construct an dictionary mapping channel names to their ids.")
+        request = self.slack_client.api_call("conversations.list")
+        if request['ok']:
+            channel_name_to_id_dictionary = {}
+            channel_id_to_name_dictionary = {}
+            for c in request['channels']:
+                channel_name_to_id_dictionary[c['name']] = c['id']
+                channel_id_to_name_dictionary[c['id']] = c['name']
+            logger.info('Constructed dictionaries containing information of {} channels.'.format(str(len(request['channels']))))
+            return channel_name_to_id_dictionary, channel_id_to_name_dictionary
+
+        else:
+            return None, None
+
+    def at_operator(self, channel):
+        '''
+        Send a Slack message to @ the operator (@ temporary operator(s) if exists, otherwise @ the one listed on google calendar.)
+        channel: id of the channel where the message will be sent
+        '''
+        if len(self.temporary_operator_id) != 0:
+            logger.info('Found temporary operator(s)!')
+            message = ""
+            for operator_id in self.temporary_operator_id:
+                message += "<@" + operator_id + "> "
+            self.send_message(channel, message)
```

Comment:
This implementation means if a temp operator is available, the normal operator is bypassed

I propose changing this to match the current "old" behavior - the "current_operator" is always paged, if temp operator(s) exist they are also paged

Reply:
Fixed :)